testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py

```diff
@@ -484,3 +485,75 @@ def test_replication_delete_event(params_from_base_test_setup, num_of_docs):
+    [None, None],
+    [load_from_data_dir, ["Sample_Image_30mb.jpg"]],
+    [generate_2_png_100_100, None],
+])
+def test_push_replication_for_20mb_doc(params_from_base_test_setup, attachment_generator, attachment_file_list):
+    """
+    @summary:
+    1. Create docs in CBL and replicate to SG using push one-shot replication
+    2. Add update to SG and CBL to create conflict
+    3. start push one-shot replication and start replication event listener
+    4. Check the error is thrown in replication event changes
+    """
+    sg_db = "db"
+    sg_admin_url = params_from_base_test_setup["sg_admin_url"]
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    sg_blip_url = params_from_base_test_setup["target_url"]
+    base_url = params_from_base_test_setup["base_url"]
+    sg_config = params_from_base_test_setup["sg_config"]
+    db = params_from_base_test_setup["db"]
+    cbl_db = params_from_base_test_setup["source_db"]
+    sync_gateway_version = params_from_base_test_setup["sync_gateway_version"]
+
+    if sync_gateway_version < "2.5.0":
+        pytest.skip('This test cannnot run with sg version below 2.5')
+    username = "autotest"
+    password = "password"
+    channels = ["ABC"]
+
+    # Create CBL database
+    sg_client = MobileRestClient()
+
+    # Reset cluster to ensure no data in system
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+
+    sg_client.create_user(url=sg_admin_url, db=sg_db, name=username, password=password, channels=channels)
+    cookie, session = sg_client.create_session(url=sg_admin_url, db=sg_db, name=username)
+    sync_cookie = "{}={}".format(cookie, session)
+    session_header = {"Cookie": sync_cookie}
+
+    # 1. Creating Docs in CBL
+    db.create_bulk_docs(number=1, id_prefix="cbl_docs", db=cbl_db, channels=channels,
+                        generator="20mb", attachments_generator=attachment_generator,
+                        attachment_file_list=attachment_file_list)
+
+    replicator = Replication(base_url)
+    repl_config = replicator.configure(source_db=cbl_db,
+                                       target_url=sg_blip_url,
+                                       continuous=False,
+                                       headers=session_header)
+    repl = replicator.create(repl_config)
+    repl_change_listener = replicator.addReplicatorEventChangeListener(repl)
+
+    # 3. Starting Replication and waiting for it finish
+    replicator.start(repl)
+    replicator.wait_until_replicator_idle(repl, max_times=5000)
+
+    # 4. Getting changes from the replication event listener
+    doc_repl_event_changes = replicator.getReplicatorEventChanges(repl_change_listener)
+    replicator.removeReplicatorEventListener(repl, repl_change_listener)
+    replicator.stop(repl)
+
+    # Processing received events
+    replicated_event_changes = get_event_changes(doc_repl_event_changes)
+    print replicated_event_changes
```

Comment:
it's better to use log_info instead of print here, print can be used on local debugging.

Reply:
removed print, as not needed.